### PR TITLE
Search by ingredient page

### PR DESCRIPTION
**Pull request by maryamkarimi**
Added search by ingredient page

Screen Shot 2020-11-22 at 9 28 35 PM
Screen Shot 2020-11-22 at 9 28 23 PM

### CODE REVIEW

**Review by jlee3227 — APPROVED** [review]

File: client/package.json [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
+    "@ant-design/icons": "^4.3.0",
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
     "antd": "^4.8.0",
+    "prop-types": "^15.7.2",
     "react": "^17.0.1",
     "react-bootstrap": "^1.4.0",
     "react-dom": "^17.0.1",
+    "react-icons": "^3.11.0",
```
Comment (jerryiu424) [inline client/package.json:15]: NIT: Remove unused dependency
```suggestion
```

File: client/package.json [hunk 2]
```diff
@@ -3,13 +3,16 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@ant-design/icons": "^4.3.0",
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
     "antd": "^4.8.0",
+    "prop-types": "^15.7.2",
```
Comment (jerryiu424) [inline client/package.json:11]: NIT: This is nice to have but would only be effective if we enforce proptypes checking on all components. Not sure if that's the direction that we want to move forward but I guess we can keep it here for now?

File: client/src/App.jsx [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+import { getRecipeByIngredientsService } from './service/recipe';
 
 function App() {
-    const [recipes, setRecipes] = useState([])
-    useEffect(() => {
-        getRecipeByIngredientsService(['apple', 'flour'], 2).then(r => setRecipes(r))
-    }, [])
+  const [recipes, setRecipes] = useState([]);
+  useEffect(() => {
+    getRecipeByIngredientsService(['apple', 'flour'], 2).then((r) => setRecipes(r));
```
Comment (jerryiu424) [inline client/src/App.jsx:9]: I'm assuming the `['apple', 'flour']` is just a temporary placeholder for now?

File: client/src/service/index.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
-export const getApiUrl = () => process.env.REACT_APP_API_URL;
+export const getApiUrl = () => 'spoonacular-recipe-food-nutrition-v1.p.rapidapi.com';
 
-export const getApiKey = () => process.env.REACT_APP_API_KEY;
\ No newline at end of file
+export const getApiKey = () => '###';
```
Comment (jerryiu424) [inline client/src/service/index.js:3]: LOL gave up on setting up the env var eh ;)

**Review by jerryiu424 — COMMENTED** [review]
Overall, code-wise looks good, Just some nits. For future improvement, maybe we can add a "drag and drop" feature, where users can drag and drop ingredients from the side bar to the main area?

File: client/package.json (hunk 1 above)
Comment (maryamkarimi) [reply client/package.json:15]: It’s used for my previous pull request, forgot to push it there.

File: client/src/App.jsx (hunk 3 above)
Comment (maryamkarimi) [reply client/src/App.jsx:9]: This is an existing class that I only re-formatted, and we would be deleting it eventually.

File: client/src/service/index.js (hunk 4 above)
Comment (maryamkarimi) [reply client/src/service/index.js:3]: Yup lol

File: client/package.json (hunk 2 above)
Comment (maryamkarimi) [reply client/package.json:11]: I guess it’s up to people to add it; but I personally would. Makes it easier to understand what each prop is

**Pull request merged**